lib/datasource/cdnjs/types.ts

```diff
@@ -1,14 +1,42 @@
+import { z } from 'zod';
+
 interface CdnjsAsset {
   version: string;
   files: string[];
-  sri?: Record<string, string>;
+  sri: Record<string, string>;
 }
 
-export interface CdnjsResponse {
+interface CdnjsResponse {
   homepage?: string;
   repository?: {
-    type: 'git' | unknown;
     url?: string;
   };
-  assets?: CdnjsAsset[];
+  assets: CdnjsAsset[];
+}
+
+/**
+ * See: https://cdnjs.com/api
+ */
+export const Response = z.object({
+  homepage: z.string().url().optional(),
+  repository: z
+    .object({
+      url: z.string().url().optional(),
+    })
+    .optional()
+    .nullable(),
+  assets: z.array(
+    z.object({
+      version: z.string(),
+      files: z.array(z.string()),
+      sri: z.record(z.string()),
+    })
+  ),
+});
+
+// export type IResponse = z.infer<typeof Response>;
```

Comment:
Once strict nulls are enabled, we'll have single source of truth for data validation and type definitions.

Reply:
However, we already can use the validation itself for runtime guarantees.